### ISSUE TITLE
Multiple accounts?

### ISSUE THREAD

**Issue by Thatoo** [vector-im/element-x-ios#546]
Hopefully, both ElementX on iOS and Android will be the first Element clients to support Multiple accounts feature (and the web client about that time too?)

Just for the record, it is one of the oldest and a wanted feature in Element-ios,    https://github.com/vector-im/element-ios/issues/315, Element-android, https://github.com/vector-im/element-android/issues/26 [and also Element-web but that's out of topic... https://github.com/vector-im/element-web/issues/2320 ].

**Comment by tazihad** [comment]
Where can I download beta version of ElementX?

**Comment by stefanceriu** [comment]
> Where can I download beta version of ElementX?

You cannot at the moment but we're actively working on a publicly available release. Otherwise I will delete your comment as it doesn't belong in this ticket

**Comment by vollkorntomate** [comment]
Am I wrong or do the Rust SDK and Element X already *almost* support multiple accounts?
I had a quick naive look into the code and from my understanding, there is nothing in the way of simply creating a second (third, ...) user session through the app, and all the storage etc. is already taken care of.

The only thing that doesn't seem to be possible yet is to restore a session for a given userID, but that looks to me like a rather small change in the `UserSessionStoreProtocol` and its implementation. And of course, the UIs for the account management are still missing, but that should be quite straightforward.

Another issue that I can think of is notification management, but I'd view that as a second step once we are able to add multiple accounts.

**Comment by ara4n** [comment]
you are right. it "just" needs the auth, login, registration, acct management, notif, account switching UI, etc. we are also desperate to have it as a feature tho; it's been excruciating not having it in classic Element.
